### ISSUE TITLE
Take ebs costs into consideration

### ISSUE THREAD

**Issue by runningman84** [aws/karpenter#5157]
### Description

**What problem are you trying to solve?**
We are going to use instance storage for container images and other stuff. Imagine this scenario:

instance c5a costs 40 USD per month
- root volume 50 GB x 1.00 instance months x 0.08 USD = 4.00 USD (EBS Storage Cost)
- data volume 50 GB x 1.00 instance months x 0.08 USD = 4.00 USD (EBS Storage Cost)
=> 50 USD

Instance c5ad costs 41 USD per month
- root volume 50 GB x 1.00 instance months x 0.08 USD = 4.00 USD (EBS Storage Cost)
- data volume based on instance store
=> 46 USD

It would be great if karpenter would take into account the given ebs costs.

We would just have two nodepools one with instance storage support and another with no instance storage (but a second ebs volume). 

**How important is this feature to you?**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
> It would be great if karpenter would take into account the given ebs costs

Are you RAID-0-ing your instance storage so that this storage can be used as ephemeral storage?

**Comment by runningman84** [comment]
Yes I am creating a raid 0 and mount that in the containerd/kubelet path in the bottlerocket instance. The volume is used for docker images emptydir and other stuff.

I will have two node pools one with instance storage and one without.

Without the ebs costs consideration in karpenter I might get instances without instance store which are eventually more expensive due to having an additional data volume.

I just suggest that you look into the block device mapping and sum up all ebs costs and add them to your general calculation.

I have seen the static volume storage pr or but I don’t think that having the costs added would mess up in any way here.
